sporkfish/endgame_tablebases/lila_tablebase.py

```diff
@@ -2,28 +2,37 @@
 from typing import Optional
 
 import chess
-import chess.syzygy
 import requests
 
+from sporkfish.endgame_tablebases.endgame_tablebase import EndgameTablebase
 
-class LilaTablebase:
-    """Class for handling the lila endgame tablebase in chess engines."""
 
-    base_url = "http://tablebase.lichess.ovh/standard?fen="
+class LilaTablebase(EndgameTablebase):
+    """
+    Class for handling the lila endgame tablebase in chess engines.
 
-    def query_bestmove(board_fen: str) -> Optional[chess.Move]:
+    :ivar BASE_URL: The base URL for the lila endgame tablebase service.
+
+    :method __init__() -> None: Initializes the LilaTablebase instance.
+    :method query(board_fen: str) -> Optional[chess.Move]: Queries the tablebase service for the best move given a board position in FEN format.
+    """
+
+    BASE_URL = "http://tablebase.lichess.ovh/standard?fen="
+
+    def __init__(self) -> None:
+        EndgameTablebase.__init__(self)
+
+    def query(self, board_fen: str) -> Optional[chess.Move]:
```

Comment:
[Design] Sorry I missed this earlier on. I'm slightly surprised this runs - pitfalls of dynamic language like python. In C++ this wouldn't compile since the base class signature query(self, board) is different from derived class query(self, board_fen)). To be consistent, could we change to query(self, board) please? Just have to call the board_fen() inside the function instead.